lib/ansible/modules/clustering/consul.py

```diff
@@ -397,8 +397,8 @@ def parse_service(module):
             module.params.get('service_port'),
             module.params.get('tags'),
         )
-    elif module.params.get('service_port') and not module.params.get('service_name'):
-        module.fail_json(msg="service_port supplied but no service_name, a name is required to configure a service.")
+    elif not module.params.get('service_name'):
```

Comment:
You maybe able to extra/better validation this with the built in option validations.

I've got some draft docs for this https://github.com/gundalow/ansible/blob/docs-argspec/docs/docsite/rst/dev_guide/developing_modules_general.rst#main-and-ansiblemodule-argument-spec

For other examples have a look at existing modules

Reply:
Hi, 
That can be done in a follow up Pull Request when you have time. Thanks again for for help module.